### PR TITLE
feat(dapp-console): Implement header in dapp console

### PR DESCRIPTION
**Pull request by imranjami**
### Description
- Implementing the header component in the `dapp-console` app
- Along with this I also made theming functional in the application with `next-themes` as per the shadcn documentation [here](https://ui.shadcn.com/docs/dark-mode/next)

### CODE REVIEW

**Comment by imranjami** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/65?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

Current dependencies on/for this PR:
* `main`
  * **PR #59** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/59?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #65** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/65?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/ecosystem/65?utm_source=stack-comment).

File: apps/dapp-console/app/layout.tsx [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
+import { PrivyProviderWrapper } from '@/app/providers/PrivyProviderWrapper'
+import { Header } from '@/app/components/Header'
+import { ThemeProvider } from './providers/ThemeProvider'
 
 const inter = Inter({ subsets: ['latin'] })
 
 export const metadata: Metadata = {
-  title: 'Create Next App',
-  description: 'Generated by create next app',
+  title: 'Supercain Developer Console',
```
Comment (jakim929) [inline apps/dapp-console/app/layout.tsx:11]: nit: typo here

**Review by jakim929 — APPROVED** [review]
lgtm!

File: apps/dapp-console/app/components/Header/index.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+import Image from 'next/image'
+import { ThemeToggle } from '@/app/components/Header/ThemeToggle'
+import { Separator } from '@eth-optimism/ui-components/src/components/ui/separator'
+import { Button } from '@eth-optimism/ui-components/src/components/ui/button'
+
+const Header = () => {
+  return (
+    <div className="h-20 px-6 border-b border-border flex items-center justify-between">
+      <div className="flex align-center gap-12">
```
Comment (jakim929) [inline apps/dapp-console/app/components/Header/index.tsx:11]: I think it's `items-center` not `align-center`

File: apps/dapp-console/app/components/Header/index.tsx [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+        <Separator orientation="vertical" className="h-6 mx-4" />
+        <Button>Sign in</Button>
+      </div>
+    </div>
+  )
+}
+
+const HeaderLogo = () => {
+  return (
+    <div className="flex align-center">
```
Comment (jakim929) [inline apps/dapp-console/app/components/Header/index.tsx:26]: `items-center`

File: apps/dapp-console/app/components/Header/ThemeToggle.tsx [hunk 4]
```diff
@@ -0,0 +1,25 @@
+'use client'
+
+import * as React from 'react'
+import { RiMoonLine, RiSunLine } from '@remixicon/react'
+import { useTheme } from 'next-themes'
+
+import { Button } from '@eth-optimism/ui-components/src/components/ui/button'
```
Comment (jakim929) [inline apps/dapp-console/app/components/Header/ThemeToggle.tsx:7]: rc:

can we change update all imports from ui-components to be 

```
import { Button } from '@eth-optimism/ui-components
```

**Pull request merged**